From 16faf3f1a47c3318ae5513d9c552c777677f4446 Mon Sep 17 00:00:00 2001
From: Victor Goldenshtein <victorg@ti.com>
Date: Wed, 26 Dec 2012 10:16:26 +0200
Subject: [PATCH 06/33] mac80211: handle iface_work in case of hw_scan

In case of hw scan, there is no reason to defer
the interface work (e.g. answering ADDBA), as
the device should be able to perform both, and
take care of the scheduling internally.

Signed-off-by: Victor Goldenshtein <victorg@ti.com>
---
 backports/net/mac80211/iface.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backports/net/mac80211/iface.c b/backports/net/mac80211/iface.c
index 573813b..2d47aee 100644
--- a/backports/net/mac80211/iface.c
+++ b/backports/net/mac80211/iface.c
@@ -1202,7 +1202,7 @@ static void ieee80211_iface_work(struct work_struct *work)
 	if (!ieee80211_sdata_running(sdata))
 		return;
 
-	if (local->scanning)
+	if (local->scanning && !local->ops->hw_scan)
 		return;
 
 	if (!ieee80211_can_run_worker(local))
-- 
2.5.1

